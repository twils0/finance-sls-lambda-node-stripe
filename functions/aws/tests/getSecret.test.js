const AWS = require('aws-sdk'); // eslint-disable-line

const { getSecret } = require('../getSecret');

const getSecretValue = jest.fn();

AWS.SecretsManager = jest.fn(() => ({
  getSecretValue,
}));

const key = 'testKey';

describe('functions', () => {
  describe('aws', () => {
    describe('getSecret', () => {
      afterEach(() => {
        getSecretValue.mockReset();
      });

      it('fails and returns callback error', async () => {
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        getSecretValue.mockImplementation((params, callback) => {
          expect(params.SecretId).toEqual(key);
          callback(error);
        });

        try {
          await getSecret(key);
        } catch (errorCatch) {
          expect(errorCatch).toEqual(error);
        }
      });

      it('fails and returns unexpected error, missing secret string', async () => {
        const emptyResult = {};
        getSecretValue.mockImplementation((params, callback) => {
          expect(params.SecretId).toEqual(key);
          callback(null, emptyResult);
        });

        try {
          await getSecret(key);
        } catch (errorCatch) {
          expect(errorCatch).toMatchSnapshot();
        }
      });

      it('correctly returns secret value', async () => {
        const secretString = {
          key1: 'value1',
          key2: 'value2',
        };
        const result = {
          SecretString: JSON.stringify(secretString),
        };
        getSecretValue.mockImplementation((params, callback) => {
          expect(params.SecretId).toEqual(key);
          callback(null, result);
        });

        const { value } = await getSecret(key);

        expect(value).toEqual(secretString);
      });
    });
  });
});
