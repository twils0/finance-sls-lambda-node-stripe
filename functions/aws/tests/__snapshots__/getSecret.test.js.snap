// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions aws getSecret fails and returns unexpected error, missing secret string 1`] = `"unexpected error - missing secret string"`;
